dpnp/backend/kernels/dpnp_krnl_common.cpp

```diff
@@ -35,6 +35,42 @@
 namespace mkl_blas = oneapi::mkl::blas;
 namespace mkl_lapack = oneapi::mkl::lapack;
 
+template <typename _DataType, typename _TypeNew, typename _ResultType>
+class dpnp_astype_c_kernel;
+
+template <typename _DataType, typename _TypeNew, typename _ResultType>
+void dpnp_astype_c(void* array1_in, void* result1, size_t size)
+{
+    cl::sycl::event event;
+
+    _DataType* array_in = reinterpret_cast<_DataType*>(array1_in);
+    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
```

Comment:
```suggestion
void dpnp_astype_c(const void* array1_in, void* result1, const size_t size)
{
    cl::sycl::event event;

    const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
    _ResultType* result = reinterpret_cast<_ResultType*>(result1);
```
Try not to ignore suggestions from previous PRs.